### DISCUSSION TITLE
Error while processing archive.

### DISCUSSION THREAD

**Discussion by swithun-liu** [Difegue/LANraragi#998] [Q&A]
**LRR Version and OS**
Please tell which version you're running, as well as your OS and installation method. (Windows, Docker, Native)

Docker (arm chip)

**Bug Details**
Please detail the bug here. 

 Error while processing archive.
(The file couldn't be moved to your content folder!)

**Matching Logs**
Post any logs you have encountered when reproducing the bug.  
If possible, switch the app to Debug Mode in order to have more logs.

**Screenshots**
<img width="1248" alt="image" src="https://github.com/Difegue/LANraragi/assets/42160297/f5451c55-4a90-4bea-a264-9afe78633a2b">

**Comment by Difegue** [comment]
Need more logs/info here I'm afraid - check write permissions on the content folder maybe

**Comment by DDzwiedziu** [comment]
I have the same problem when installing from Docker compose. If you use bind mounts, as described in the example [1] then the directories are mounted with the ownership of the user running the docker commands. And if you use the Docker engine to create the volumes, then the owner is either root:root or koyomi:root.

From inside the container:
```
/home/koyomi/lanraragi # ls -hal
total 168K   
drwxr-sr-x    1 koyomi   koyomi      4.0K Dec  9 13:24 .
drwxr-sr-x    1 koyomi   koyomi      4.0K Dec  9 13:24 ..
-rw-r--r--    1 root     koyomi        52 Nov  4 00:26 config.log
drwxr-xr-x    3 root     root        4.0K Dec  9 13:24 content
drwxr-xr-x    3 koyomi   root        4.0K Dec  9 13:24 database
drwxr-sr-x    1 koyomi   koyomi      4.0K Nov  4 00:17 lib
drwxr-sr-x    2 koyomi   koyomi      4.0K Dec  9 13:24 log
-rw-r--r--    1 koyomi   koyomi       186 Nov  4 00:17 lrr.conf
-rw-r--r--    1 koyomi   koyomi     97.7K Nov  4 00:17 package-lock.json
-rw-r--r--    1 koyomi   koyomi      2.1K Nov  4 00:17 package.json
drwxr-sr-x    1 root     koyomi      4.0K Dec  9 13:24 public
drwxr-sr-x    2 koyomi   koyomi      4.0K Nov  4 00:17 script
drwxr-sr-x    3 koyomi   koyomi      4.0K Nov  4 00:17 templates
drwxr-sr-x    4 koyomi   koyomi      4.0K Nov  4 00:17 tests
drwxr-sr-x    1 koyomi   koyomi      4.0K Nov  4 00:27 tools
```

From the host
```
~dzwiedziu/ ls -hal /home/docker-data/volumes/lanr*
/home/docker-data/volumes/lanraragi_content:
total 12K
drwx-----x  3 root root 4,0K 12-09 14:43 ./
drwx-----x 43 root root 4,0K 12-09 14:43 ../
drwxr-xr-x  3 root root 4,0K 12-09 14:43 _data/

/home/docker-data/volumes/lanraragi_db:
total 12K
drwx-----x  3 root root 4,0K 12-09 14:43 ./
drwx-----x 43 root root 4,0K 12-09 14:43 ../
drwxr-xr-x  3 9001 root 4,0K 12-09 14:43 _data/
```

I've chowned both `_data` to 9001: 

```
~dzwiedziu/ chown -R 9001 /home/docker-data/volumes/lanr*/_data
~dzwiedziu/ ls -hal /home/docker-data/volumes/lanr*
/home/docker-data/volumes/lanraragi_content:
total 12K
drwx-----x  3 root root 4,0K 12-09 14:43 ./
drwx-----x 43 root root 4,0K 12-09 14:43 ../
drwxr-xr-x  3 9001 root 4,0K 12-09 14:43 _data/

/home/docker-data/volumes/lanraragi_db:
total 12K
drwx-----x  3 root root 4,0K 12-09 14:43 ./
drwx-----x 43 root root 4,0K 12-09 14:43 ../
drwxr-xr-x  3 9001 root 4,0K 12-09 14:43 _data/
```

…and it worked. Just this might need adjusting at the Dockerfile level (did not take a deeper look into it yet).

[1] https://sugoi.gitbook.io/lanraragi/dev/installing-lanraragi/docker -> `docker-compose.yml`
